cc/ccutils.js

```diff
@@ -525,9 +527,27 @@ function getTransactionsMany(peers, mypk, ...args)
     const parsedInTransactions = {}
 
     if (inTransactionsIds.length > 0) {
-      inTransactionsIds = inTransactionsIds.flat()
-      let inTransactions = await getTransactionsMany(peers, mypk, ...inTransactionsIds);
-      inTransactions.transactions.forEach(tx => {
+      inTransactionsIds = inTransactionsIds.flat();
+
+      let inTransactionsFull;
+
+      if (inTransactionsIds.length > MAX_TX_PER_REQUEST) {
+        const promises = [];
+        const chunks = splitArrayInChunks(inTransactionsIds, MAX_TX_PER_REQUEST);
+        chunks.forEach((chunk, idx) => 
+          setTimeout(() => {
+            return promises.push(getTransactionsMany(peers, mypk, ...chunk)) 
+          }, 100 * idx)
+        );
+        let promiseResults = await Promise.all(promises);
+        promiseResults = promiseResults.map(tx => tx.transactions);
+        inTransactionsFull = promiseResults.flat();
```

Comment:
if I understand correctly, Promise.all will run as promises are still being added to the `promises` array. Not sure if this is an anti-pattern or not, but looks a bit confusing to me.

Maybe we can resolve each promise in the `forEach` and add the result to `inTransactionsFull`, or even do some clever usage of `reduce`:

```suggestion
        inTransactionsFull = await chunks.reduce(async (previousPromise, chunk, index) => {
          let transactions = await previousPromise;

          const tx = await new Promise((resolve) => setTimeout(() => {
            getTransactionsMany(peers, mypk, ...chunk).then((value) => resolve(value));
          }, 100 * index));

          return [...transactions, ...tx.transactions];
        }, Promise.resolve([]));
```